cargo-dist/src/init.rs

```diff
@@ -84,15 +128,59 @@ pub fn do_init(cfg: &Config, args: &InitArgs) -> DistResult<()> {
         }
     };
 
+    // If we're migrating, the configuration will be missing the
+    // generic workspace specification, and will have some
+    // extraneous cargo-specific stuff that we don't want.
+    if is_migrating {
+        // Remove Cargo-specific workspace
+        workspace_toml.remove("workspace");
+
+        // Write generic workspace config
+        let mut table = toml_edit::table();
+        if let Some(t) = table.as_table_mut() {
+            let mut array = toml_edit::Array::new();
+            array.push("cargo:.");
+            t["members"] = toml_edit::value(array);
+            t.set_position(0);
+        }
+        workspace_toml.insert("workspace", table);
+
+        // These fields will continue to be read from Cargo.toml
+        workspace_toml.remove("dependencies");
+        workspace_toml.remove("dev-dependencies");
+        workspace_toml.remove("package");
+        workspace_toml.remove("profile");
+    }
+
     if let Some(meta) = &multi_meta.workspace {
-        apply_dist_to_workspace_toml(&mut workspace_toml, root_workspace.kind, meta);
+        apply_dist_to_workspace_toml(&mut workspace_toml, desired_workspace_kind, meta);
     }
 
     eprintln!();
 
+    let filename = match desired_workspace_kind {
+        WorkspaceKind::Rust => "Cargo.toml",
+        WorkspaceKind::Generic => "dist-workspace.toml",
+        WorkspaceKind::Javascript => todo!(),
+    };
+    let destination = root_workspace.workspace_dir.join(filename);
+
     // Save the workspace toml (potentially an effective no-op if we made no edits)
-    config::save_cargo_toml(&root_workspace.manifest_path, workspace_toml)?;
-    eprintln!("{check} added [workspace.metadata.dist] to your root Cargo.toml");
+    config::save_cargo_toml(&destination, workspace_toml)?;
+    eprintln!("{check} added [workspace.metadata.dist] to your root {filename}");
```

Comment:
This label is incorrect now (it's just `dist` in the dist-workspace.toml, right?)